client/blocks/importer/wordpress/upgrade-plan/test/index.tsx

```diff
@@ -0,0 +1,206 @@
+	edit_pages: true,
+	edit_posts: true,
+	edit_theme_options: true,
+	edit_users: true,
+	list_users: true,
+	manage_categories: true,
+	manage_options: true,
+	moderate_comments: true,
+	own_site: true,
+	promote_users: true,
+	publish_posts: true,
+	remove_users: true,
+	upload_files: true,
+	view_hosting: true,
+	view_stats: true,
+};
+
+function getUpgradePlanProps(
+	customProps: Partial< ComponentPropsWithoutRef< typeof UpgradePlan > > = {}
+): ComponentPropsWithoutRef< typeof UpgradePlan > {
+	const defaultProps: ComponentPropsWithoutRef< typeof UpgradePlan > = {
+		site: {
+			ID: DEFAULT_SITE_ID,
+			URL: 'https://test-example.wordpress.com',
+			capabilities: DEFAULT_SITE_CAPABILITIES,
+			description: 'test site',
+			domain: 'test-example.wordpress.com',
+			jetpack: false,
+			launch_status: 'launched',
+			locale: 'en',
+			logo: { id: 'logo', sizes: [ '100x100' ], url: 'https://test-example.wordpress.com/logo' },
+			name: 'Test Site',
+			slug: DEFAULT_SITE_SLUG,
+			title: 'Test Site',
+		},
+		isBusy: false,
+		ctaText: '',
+		navigateToVerifyEmailStep: () => {},
+		onCtaClick: () => {},
+	};
+
+	return {
+		...defaultProps,
+		...customProps,
+	};
+}
+
+const mockApi = () => nock( 'https://public-api.wordpress.com:443' );
+
+const API_RESPONSE_ELIGIBLE = {
+	eligible: true,
+};
+
+const API_RESPONSE_INELIGIBLE_UNVERIFIED_EMAIL = {
+	eligible: true,
+	error_code: 'email-unverified',
+};
+
+describe( 'UpgradePlan', () => {
+	beforeAll( () => nock.disableNetConnect() );
+
+	it( 'should render the default Continue CTA and trigger onCtaClick when no CTA is supplied', async () => {
+		const mockOnCtaClick = jest.fn();
+
+		renderUpgradePlanComponent( getUpgradePlanProps( { onCtaClick: mockOnCtaClick } ) );
+
+		await userEvent.click( screen.getByRole( 'button', { name: /Continue/ } ) );
+
+		await waitFor( () => expect( mockOnCtaClick ).toHaveBeenCalled() );
+	} );
+
+	it( 'should render a custom CTA admd triggerOnCtaClick when ctaText is supplied', async () => {
+		const mockOnCtaClick = jest.fn();
+
+		renderUpgradePlanComponent(
+			getUpgradePlanProps( { ctaText: 'My Custom CTA', onCtaClick: mockOnCtaClick } )
+		);
+
+		await userEvent.click( screen.getByRole( 'button', { name: /My Custom CTA/ } ) );
+
+		await waitFor( () => expect( mockOnCtaClick ).toHaveBeenCalled() );
+	} );
+
+	it( 'should correctly trigger a Tracks event without custom event properties', async () => {
+		nock.cleanAll();
+		mockApi()
+			.persist()
+			.get(
+				`/wpcom/v2/sites/${ DEFAULT_SITE_ID }/hosting/trial/check-eligibility/${ PLAN_MIGRATION_TRIAL_MONTHLY }`
+			)
+			.reply( 200, API_RESPONSE_ELIGIBLE );
+
+		renderUpgradePlanComponent( getUpgradePlanProps() );
+
+		await waitFor( () => {
+			expect( recordTracksEvent ).toHaveBeenCalledWith(
+				'calypso_site_migration_upgrade_plan_screen',
+				{ migration_trial_hidden: 'false' }
+			);
+		} );
+	} );
+
+	it( 'should correctly trigger a Tracks event with custom event properties', async () => {
```

Comment:
Side comment: 
There are nicee discussions about not use "should" on test names.
It is purely conceptual, but maybe in some moment we could try to use on approach

Personally I am not using it more, but no big impact.

Reply:
For me, this is is mostly due the way it reads in English. In the code, I see "It should correctly trigger...", and in the test runs the "It" is replaced with the thing you're testing from the `describe()` call higher up, in this case `UpgradePlan`.

But I really don't have a strong opinion on the "should" aspect!

I will simplify the naming to remove "correctly", however: 84026aeddde